woocommerce/woocommerce-gutenberg-products-block
# Include Cart and Checkout Blocks when included in WC Core

This PR will graduate the Cart and Checkout Blocks so they are available even when WooCommerce Blocks is not running as a feature plugin (i.e. it is included in WooCommerce Core).

To achieve this, I ensured both blocks, and their inner blocks were registered using `registerBlockType` rather than `registerFeaturePluginBlockType` - `registerFeaturePluginBlockType` has been commented out, but left in the `feature-flags.ts` file so it can easily be reinstated in future if we ever add a feature-plugin-only block.

I then found all instances in PHP where classes were only registered if WooCommerce Blocks was running as a feature plugin (Payments Api was the only place).

There was also an inbox notification added in #4518 to surface these blocks to a certain subset of users, however this message is defunct. I removed the code to _add_ this notification, but left the code to _delete_ it. This should ensure any sites that still have this message have it removed.



Fixes #6699




#### Other Checks


### Testing

#### Automated Tests

#### User Facing Testing



1. Run `npm run build`.
2. Replace the content of`blocks.ini` with the following config:
```
woocommerce_blocks_phase = 1
woocommerce_blocks_env = production
```
3. See the Cart and Checkout Blocks and ensure they work as expected. Testing for each block is below:

#### Cart 
1. In the Cart Block in the editor, ensure the shipping calculator is enabled.
5. Enable coupons on your site (WooCommerce -> Settings -> General ->  Enable the use of coupon codes) and add one.
6. Add two shipping methods for different locations, e.g. two for USA and two for UK.
7. **Do not** add items to your cart.
8. Go to the Cart Block and ensure the empty cart is displayed.
9. Add items to your cart and revisit the Cart Block.
12. Enter your address in the shipping calculator, ensure the correct shipping prices are displayed based on your location.
13. Enter a coupon and ensure it works, and the price is modified accordingly.
14. Modify the quantities of items in your cart, ensure it works and the prices update properly.
15. If you have express payments enabled and working, try checking out _from_ the Cart Block using GPay or Apple pay.
16. If you don't, hit proceed to checkout and ensure it works correctly.

#### Checkout

1. Go to WooCommerce -> Settings -> Accounts & Privacy and enable ` Allow customers to create an account during checkout` and `Allow customers to log into an existing account during checkout`.
2. Enable the Stripe payment method, COD, and cheque.
5. Log out of the site then add items to your cart and go to the Checkout Block. 
6. Check the `create account` box when entering your details.
7. Remove the coupon added in the Cart testing steps, and try re-adding it.
8. Change shipping methods and ensure the prices update.
9. Change address, ensure shipping prices update when the country changes.
10. Leave some fields empty on the address form, ensure you get an error.
11. Re-fill them and enter an invalid ZIP code. (Set country to UK and enter 234345 as the Postcode). Ensure you see an error at the top of the block.
13. Correct the error (A valid one is `SW19 1AA` if in UK).
16. Check out and ensure it works and the order is logged correctly.
17. Ensure you received emails relating to your order (if using Local WP you can go to your site -> tools -> mailhog to check this).

### Manual testing for Rubik only
After trying all the steps above, please:

1. From your `plugins` directory, run: `npx @wordpress/create-block -t @woocommerce/extend-cart-checkout-block checkout-integration`. 
2. Activate the `Checkout Integration` plugin in your dashboard.
3. Go to the Cart Block.
4. Ensure the names of the items in your cart contain `+ extra data!` (this is from the `checkout integration` plugin).
5. Go to the Checkout Block.
14. Change your city to `Denver`. Ensure the COD option disappears from available payment methods.
15. Check to make sure there is a block at the bottom of the Checkout Order Summary it should say `Data passed to this component: This is some example data from the server` - this signifies Slot/Fill is working.


### WooCommerce Visibility




### Changelog

> Enable the Cart and Checkout blocks when WooCommerce Blocks is bundled in WooCommerce Core.
